samtranslator/validator/value_validator.py

```diff
@@ -17,13 +17,20 @@ class ExpectedType(Enum):
 
 class _ResourcePropertyValueValidator(Generic[T]):
     value: Optional[T]
-    resource_logical_id: str
+    resource_logical_id: Optional[str]
+    event_id: Optional[str]
     property_identifier: str
 
-    def __init__(self, value: Optional[T], resource_logical_id: str, property_identifier: str) -> None:
+    def __init__(
+        self, value: Optional[T], resource_id: str, property_identifier: str, is_sam_event: bool = False
```

Comment:
What's a SAM event? The `Events` stuff?

Reply:
For event issues, there is a different exception class for that